src/java/picard/vcf/FindMendelianViolations.java

```diff
@@ -0,0 +1,301 @@
+)
+public class FindMendelianViolations extends CommandLineProgram {
+    @Option(shortName = StandardOptionDefinitions.INPUT_SHORT_NAME, doc = "Input VCF or BCF with genotypes.")
+    public File INPUT;
+
+    @Option(shortName = "PED", doc = "File of Trio information in PED format (with no genotype columns).")
+    public File TRIOS;
+
+    @Option(shortName = StandardOptionDefinitions.OUTPUT_SHORT_NAME, doc = "Output metrics file.")
+    public File OUTPUT;
+
+    @Option(shortName = "GQ", doc = "Minimum genotyping quality (or non-ref likelihood) to perform tests.")
+    public int MIN_GQ = 30;
+
+    @Option(shortName = "DP", doc="Minimum depth in each sample to consider possible mendelian violations.")
+    public int MIN_DP = 0;
+
+    @Option(shortName = "MINHET", doc = "Minimum allele balance at sites that are heterozygous in the offspring.")
+    public double MIN_HET_FRACTION = 0.3;
+
+    @Option(optional = true, doc = "If provided, output per-family VCFs of mendelian violations into this directory.")
+    public File VCF_DIR;
+
+    @Option(doc = "List of chromosome names to skip entirely.")
+    public Set<String> SKIP_CHROMS = CollectionUtil.makeSet("MT", "chrM");
+
+    @Option(doc = "List of possible names for male sex chromosome(s)")
+    public Set<String> MALE_CHROMS = CollectionUtil.makeSet("Y", "chrY");
+
+    @Option(doc = "List of possible names for female sex chromosome(s)")
+    public Set<String> FEMALE_CHROMS = CollectionUtil.makeSet("X", "chrX");
+
+    @Option(doc = "List of chr:start-end for pseudo-autosomal regions on the female sex chromosome. Defaults to HG19/b37 & HG38 coordinates.")
+    public Set<String> PSEUDO_AUTOSOMAL_REGIONS = CollectionUtil.makeSet("X:10001-2649520", "X:59034050-59373566", "chrX:10000-2781479", "chrX:155701382-156030895");
+
+    @Option(doc = "The number of threads that will be used to collect the metrics. ")
+    public int THREAD_COUNT = 1;
+
+    @Option(doc = "If true then fields need to be delimited by a single tab. If false the delimiter is one or more whitespace characters." +
+            " Note that tab mode does not strictly follow the PED spec")
+    public boolean TAB_MODE = false;
+
+    private final Log LOG = Log.getInstance(FindMendelianViolations.class);
+    private final ProgressLogger progressLogger = new ProgressLogger(LOG, 10000, "variants analyzed");
+
+    private final Lazy<VCFHeader> inputHeader = new Lazy<>(new Lazy.LazyInitializer<VCFHeader>() {
+        @Override
+        public VCFHeader make() {
+            try (final VCFFileReader in = new VCFFileReader(INPUT)) {
+                return in.getFileHeader();
+            }
+        }
+    });
+
+    private final Lazy<PedFile> pedFile = new Lazy<>(new Lazy.LazyInitializer<PedFile>() {
+        @Override
+        public PedFile make() {
+            return PedFile.fromFile(TRIOS, TAB_MODE);
+        }
+    });
+
+    private final List<Interval> parIntervals = new ArrayList<>();
```

Comment:
True - but it's only final because you initialize it to an empty list, and then fill the list at the start of operations.